data-prepper-plugins/opensearch-source/src/main/java/org/opensearch/dataprepper/plugins/source/opensearch/SourceInfo.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.source.opensearch;
+
+public class SourceInfo {
+
+    private String osVersion;
+
+    private String dataSource;
```

Comment:
Let's make these two fields final. They should not change through the lifecycle of these cluster.